src/Illuminate/Container/Container.php

```diff
@@ -940,6 +940,16 @@ public function getBindings()
     }
 
     /**
+     * Return the types which are bound in this container, except for aliases.
+     *
+     * @return array
+     */
+    public function getAbstractTypes()
+    {
+        return array_unique(array_merge(array_keys($this->bindings), array_keys($this->instances)));
```

Comment:
Fixed.

W.r.g.t. quartic time complexity, PHP docs state:

> array_unique() sorts the values treated as string at first, then will keep the first key encountered for every value, and ignore all following keys. It does not mean that the key of the first related value from the unsorted array will be kept.

That is worst-case quadratic only if all strings have the same length as the array itself. Could you point me to a reference explicating why it is quartic?